volumes/miovision/api/intersection_tmc.py

```diff
@@ -305,70 +301,125 @@ def daterange(start_time, end_time, dt):
         yield (c_start_t, c_start_t + dt)
 
 
+class IntersectionInfo:
```

Comment:
This is an excellent idea.

Was thinking that it might be a more easily interpreted if the `Intersection` class represents a single `Intersection` and then there's a class factory that returns a list. Then the instance methods would apply to singular intersection instances.

Reply:
Slightly confused - I should make `Intersection` and `IntersectionArray` classes, the former of which includes date range lookup methods and the latter of which instantiates and stores instances of the former, right? I though a class factory returns classes of different types rather than making a bunch of classes of the same type?

Reply:
Maybe factory was the wrong word. I meant a class function that returns a list of `Intersection` objects. An `IntersectionArray` class seems overkill to represent a list of `Intersection` objects IMO.